plugins/org.jkiss.dbeaver.net.ssh/src/org/jkiss/dbeaver/model/net/ssh/SSHUtils.java

```diff
@@ -78,4 +90,94 @@ public static boolean isKeyEncrypted(byte[] privKeyValue) {
+        try {
+            String userHomePathString = System.getProperty(IConstants.SYSTEM_PROPERTY_USER_HOME);
+            if (userHomePathString != null) {
+                Path userHomeDirPath = Paths.get(userHomePathString);
+                if (userHomeDirPath.toFile().isDirectory()) {
+                    Path sshHomeDirPath = userHomeDirPath.resolve(DEFAULT_SSH_HOME_DIR_NAME);
+                    File sshHomeDir = sshHomeDirPath.toFile();
+
+                    if (RuntimeUtils.isWindows() && (!sshHomeDir.isDirectory() || !sshHomeDir.exists())) {
+                        Path sshHomeOldDirPath = userHomeDirPath.resolve(DEFAULT_SSH_HOME_DIR_NAME_WIN_OLD);
+                        File sshHomeOldDir = sshHomeDirPath.toFile();
+                        if (sshHomeOldDir.isDirectory()) {
+                            sshHomeDirPath = sshHomeOldDirPath;
+                            sshHomeDir = sshHomeOldDir;
+                        }
+                    }
+
+                    if (sshHomeDir.isDirectory() || !sshHomeDir.exists()) {
+                        // don't need to create it until we'll need to write known hosts file
+
+                        if (updatePreferences) {
+                            Platform.getPreferencesService()
+                                    .getRootNode().node(PLATFORM_SSH_PREFERENCES_NODE)
+                                    .put(PLATFORM_SSH_PREFERENCES_SSH2HOME_KEY, sshHomeDir.getAbsolutePath());
+                        }
+
+                        Path knownSshHostsFilePath = sshHomeDirPath.resolve(KNOWN_SSH_HOSTS_FILE_NAME);
+                        return knownSshHostsFilePath.toFile();
+                    } else {
+                        log.warn("Failed to resolve default SSH known hosts file location due to invalid SSH home directory " + sshHomeDirPath.toAbsolutePath());
+                    }
+                } else {
+                    log.warn("Failed to resolve default SSH known hosts file location due to missing user home directory " + userHomeDirPath.toAbsolutePath());
+                }
+            } else {
+                log.warn("Failed to resolve default SSH known hosts file location due to missing user home system property.");
+            }
+        } catch (Throwable e) {
+            log.warn("Failed to resolve default SSH known hosts file location.", e);
+        }
+
+        if (forceFileObjectOnFail) {
+            Path forcedUserProfilePath = Paths.get(RuntimeUtils.isWindows() ? "%USERPROFILE%" : "~"); // let's pretend it'll resolve itself
+            return forcedUserProfilePath.resolve(DEFAULT_SSH_HOME_DIR_NAME).resolve(KNOWN_SSH_HOSTS_FILE_NAME).toFile();
+        } else {
+            return null;
+        }
+    }
+
+    public static void forcePlatformReloadKnownHostsPreferences() {
+        JSchCorePlugin.getPlugin().setNeedToLoadKnownHosts(true);
+        JSchCorePlugin.getPlugin().getJSch().setHostKeyRepository(null);
+    }
```

Comment:
Why do we even touch the platform JSCH plugin?

Reply:
Because we need to update preferences page to reload content of known-hosts file. It doesn't reload it on content changes automatically.

Reply:
I still don't get it. Is it somehow related to `PLATFORM_SSH_PREFERENCES_NODE`?

Reply:
Yes, it's related. It's about eclipse jsch plugin. 
We take preferences about user ssh dir and content of known-hosts file from it and when known-hosts file changed we want to see updated list of records in preferences page. We need to force reload it due to eclipse doesn't reread file if there is changes only in it's content.

Reply:
Makes sense.